src/services/identity/SitesService.ts

```diff
@@ -66,6 +67,89 @@ class SitesService {
     this.tokenStore = tokenStore
   }
 
+  assertGitHubCommitData(commit: any): commit is GitHubCommitData {
+    if (
+      commit &&
+      (commit as GitHubCommitData).author !== undefined &&
+      (commit as GitHubCommitData).author.name !== undefined &&
+      (commit as GitHubCommitData).author.date !== undefined &&
+      (commit as GitHubCommitData).author.email !== undefined &&
+      (commit as GitHubCommitData).message !== undefined
+    ) {
+      return true
+    }
+
+    return false
+  }
+
+  async insertUrlsFromConfigYml(
+    siteUrls: SiteUrls,
+    sessionData: UserWithSiteSessionData
+  ): Promise<SiteUrls> {
+    if (siteUrls.staging && siteUrls.prod) {
+      // We call ConfigYmlService only when necessary
+      return siteUrls
+    }
+
+    const {
+      content: configYmlData,
+    }: { content: ConfigYmlData } = await this.configYmlService.read(
+      sessionData
+    )
+
+    // Only replace the urls if they are not already present
+    siteUrls = {
+      staging:
+        configYmlData.staging && !siteUrls.staging
+          ? configYmlData.staging
+          : siteUrls.staging,
+      prod:
+        configYmlData.prod && !siteUrls.prod
+          ? configYmlData.prod
+          : siteUrls.prod,
+    }
```

Comment:
could we have a helper method for this? (cos also done below but minor)

also, we should create a new object rather than passing `siteUrls` - this modifies `siteUrl` in place and js doesn't pass by value so this is potentially dangerous

Reply:
Hmm I think it might be clearer here instead of having a separate helper method?

Also adjusted the name/assignment in 3692b08.